src/Clapi/Relay.hs

```diff
@@ -100,3 +102,37 @@ import Data.Maybe.Clapi (note)
 -- processOms u oms =
 --   do
 --     undefined
+
+applyMessage :: (MonadFail m) => Valuespace Unvalidated -> Message -> m (Valuespace Unvalidated)
+applyMessage vs msg = case msg of
+    (MsgSet p t v i a s) -> vsSet a i v p s t vs
+    (MsgAdd p t v i a s) -> vsAdd a i v p s t vs
+    (MsgRemove p t a s) -> vsRemove a p s t vs
+    (MsgClear p t a s) -> vsClear a p s t vs
+    (MsgAssignType p tp) -> return $ vsAssignType p tp vs
+    (MsgDelete p) -> vsDelete p vs
+    (MsgChildren p c) -> undefined -- FIXME: backing implementation
+    (MsgSubscribe _) -> return vs
+    (MsgUnsubscribe _) -> return vs
+    (MsgError _ _) -> return vs
+
+applyMessages :: Valuespace v -> [Message] -> MonadErrorMap (Valuespace Unvalidated)
+applyMessages mvvs = applySuccessful [] (unvalidate mvvs)
+  where
+    applySuccessful errs vs [] = (Map.fromList errs, vs)
+    applySuccessful errs vs (m:ms) = case canFailApply vs m of
+        (Left es) -> applySuccessful ((msgPath' m, es):errs) vs ms
+        (Right vs') -> applySuccessful errs vs' ms
+    canFailApply :: Valuespace Unvalidated -> Message -> CanFail (Valuespace Unvalidated)
+    canFailApply = applyMessage
+
+handleOwnerMessages :: Valuespace Validated -> [Message] -> ([Message], Valuespace Validated)
+handleOwnerMessages vs msgs = (rmsgs, rvs)
+  where
+    rvs = if errored then vs else vvs
+    errored = errs == Map.empty
```

Comment:
`errs /= Map.empty` no?

Reply:
I think seeing as `Map` is a `Foldable` you can also say `errored = not . null $ errs`

Reply:
I fixed it in a later commit, but have gone to the not null thing for tidytude.